packages/storage/src/types/inputs.ts

```diff
@@ -49,4 +49,4 @@ export type StorageCopyInput<
 /**
  * The data payload type for upload operation.
  */
-export type StorageUploadDataPayload = Blob | BufferSource | string | File;
+export type StorageUploadDataPayload = Blob | BufferSource | string;
```

Comment:
I think I found a more organic solution to make `File` type available according to the runtime:

```suggestion
export type StorageUploadDataPayload =
	| Blob
	| ArrayBufferView
	| ArrayBuffer
	| string
	| OptionalFile;

/**
 * File type is not available in environments like React Native. This type
 * returns File type only when it's defined in global.
 */
type OptionalFile = unknown extends File ? never : File;
```

The `BufferSource` is replaced by `ArrayBufferView | ArrayBuffer` explicitly and they are equivalent.

Reply:
In browser:
<img width="850" alt="Screenshot 2023-10-12 at 3 28 11 PM" src="https://github.com/aws-amplify/amplify-js/assets/7614947/7317a03f-a106-4e3a-8d14-ed2ded48d7ce">
In RN:
<img width="795" alt="Screenshot 2023-10-12 at 3 28 19 PM" src="https://github.com/aws-amplify/amplify-js/assets/7614947/865fb9ef-9a91-428f-9410-2f6bae1d4370">

Reply:
I think for removing `File` we are creating new api this change is to support `uploadFile` api. What do you think ?

Reply:
Thanks for the suggestion I will change the types and test in sample app.